napari/layers/utils/text_manager.py

```diff
@@ -51,92 +49,97 @@ class TextManager(EventedModel):
         Offset from the anchor point.
     rotation : float
         Angle of the text elements around the anchor point. Default value is 0.
+    color : ColorEncoding
+        Defines the color for each text element.
+    text : StringEncoding
+        Defines the string for each text element.
+
+    Private Attributes
+    ------------------
+    _n_text : int
+        The number of text elements managed, which should correspond to the number
+        of rows in the properties table. Should be removed if/when the properties
+        table reliably stores that count.
     """
 
-    values: Array[str] = []
+    # Declare properties as a generic dict so that a copy is not made on validation
+    # and we can rely on a layer and this sharing the same instance.
+    properties: dict
     visible: bool = True
     size: PositiveInt = 12
-    color: Array[float, (4,)] = 'cyan'
     blending: Blending = Blending.TRANSLUCENT
     anchor: Anchor = Anchor.CENTER
     # Use a scalar default translation to broadcast to any dimensionality.
     translation: Array[float] = 0
     rotation: float = 0
-    _mode: TextMode = TextMode.NONE
-    _text_format_string: str = ''
+    text: Union[STRING_ENCODINGS] = FormatStringEncoding(format_string='')
```

Comment:
This makes a lot of sense. I don't love that we end up with `layer.text.text`, but maybe we can rename `layer.text` - I'm fine to do in a later PR with its own deprecation there though

Reply:
I was thinking of renaming this as `string` (to match `StringEncoding`), so we'd have `layer.text.string` and `layer.text.color`. And then likely adding `text` as a Pydantic alias of `string` to support backwards compatibility. Thoughts?

Reply:
very nice, i like! should they be `layer.text.strings` and `layer.text.colors` i.e. plurals?

Reply:
I actually prefer the singular, again to be kind of consistent with Altair, and also because these are defining single declarative encodings (that will then be responsible for generating multiple values). This would also provide backwards compatibility with the existing `color` field.

But I don't feel strongly, and I think BC for `color` could also be achieved with an alias, so happy to change to plural.

Reply:
let's go singular - totally fine with me, and good to be consistent with other tools/ backwards compatible